libcontainer/factory_linux.go

```diff
@@ -327,6 +339,16 @@ func (l *LinuxFactory) Load(id string) (Container, error) {
 		root:                 containerRoot,
 		created:              state.Created,
 	}
+	if state.Config.ChownCgroup {
+		uid, err := state.Config.HostRootUID()
+		if err != nil {
+			return nil, err
+		}
+		err = c.cgroupManager.SetUID(&uid)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
1. Why this code needs to be in `Load`? I can't think of any code paths that can use it...

2. If this is indeed required here, maybe it's better to have a function to be called from both `Load` and `Create`.

Reply:
Latest version of patch conveys the owner UID in the `configs.Cgroup` struct.  No changes needed in `factory_linux.go` anymore.